docs/notebooks/recovering_from_errors.pct.py

```diff
@@ -124,6 +124,24 @@ def __call__(self, x):
     )
     plot_bo_points(data.query_points.numpy(), ax[0, 0], 5, arg_min_idx)
 
+# %% [markdown]
+# ## Out of memory errors
+#
+# At the moment, trieste's `OptimizationResult` doesn't allow you to recover from Out Of Memory errors, which normally result in the Python process shutting down. One possible cause of memory errors is performing a gradient-based optimization over a large dataset. To work around this, you can specify that evaluations of the acquisition function be batchified: this splits them (on the first dimension) into batches of a given size, then stitches them back together. To do this, you need to provide an explicit batchified optimizer and specify a desired batch size.
+
+# %%
+from trieste.acquisition.optimizer import (
+    automatic_optimizer_selector,
+    batchify_acquisition_function_calls,
+)
+from trieste.acquisition.rule import EfficientGlobalOptimization
+
+optimizer = batchify_acquisition_function_calls(
+    automatic_optimizer_selector, batch_size=10_000
```

Comment:
to make it more realistic perhaps something smaller than 10_000? conceivably you would use something like 5-10 rather than 10000 if you have memory issues, no?

Reply:
Yeah, batches of 5-10 would be very inefficient. Remember this is the overall size of the tensor, not the leading dimension size.

Reply:
ah, I completely misunderstood it then, I thought this was batching optimization runs, rather than predictions...much clearer now :)